burr/core/application.py

```diff
@@ -2207,9 +2207,19 @@ def with_state_persister(
         """
         if on_every != "step":
             raise ValueError(f"on_every {on_every} not supported")
+
         if not isinstance(persister, persistence.BaseStateSaver):
             self.lifecycle_adapters.append(persister)
         else:
+            # Check if 'is_initialized' exists and is False; raise RuntimeError, else continue if not implemented
+            try:
+                if not persister.is_initialized():
+                    raise RuntimeError(
```

Comment:
Might be nice to have a subclass `UnintiailizedPersisterException`, or just use `ValueError` -- while it's a bit debated, `RuntimeError` is generally a state that's specifically not supposed to happen: https://discuss.python.org/t/meaning-of-runtimeerror/56206

Reply:
I'm in this camp:

> For me, raising RuntimeError is for serious failures of implementation.
It isn’t for situations like raising ValueError, when I’ve been given a
bad value. It’s for situation which should never occur, particularly
things which imply a bad decision or action earlier.

Reply:
Yeah, I read that as more of an internal thing, E.G. not user-facing. The user has made a mistake by not initializing here -- more of a bad state error.